File: src/graphics/TLineForcePoly.js
/**
 * Created by simong on 2/20/17.
 */

import TLine from './TLine';
import styles from './styles';
import TLineMember from './TLineMember';
import util from './util';

export default class TLineForcePoly extends TLine {
    static ZERO_LENGTH = 0.3;

    constructor(graphics, startPoint, endPoint, memberStart, memberEnd, member, options = {}) {
        super(graphics, startPoint, endPoint, options);
        this.mMemberStart = memberStart;
        this.mMemberEnd = memberEnd;
        member.setForcePolyMember(this);
        this.mSize = options.size || 3;
        this.update();
    }

    update() {
        if (!this.mMemberStart || !this.mMemberEnd) {
            return;
        }
        if (this.length() <= TLineForcePoly.ZERO_LENGTH) {
            this.mCharacter = TLineMember.NONE;
            this.mColor = TLineMember.ColorZero;
        } else if (util.near(util.direction(this.mMemberStart.x, this.mMemberStart.y, this.mMemberEnd.x, this.mMemberEnd.y), this.direction(), 0.02)) {
            this.mCharacter = TLineMember.TENSILE;
            this.mColor = TLineMember.ColorTensile;
        } else {
            this.mCharacter = TLineMember.COMPRESSIVE;
            this.mColor = TLineMember.ColorCompressive;
        }
        super.update();
    }
}

File: old_java_src/ActiveStatics/src/truss/TLineForcePoly.java
package truss;

/**
 * Title:        Truss
 * Description:
 * Copyright:    Copyright (c) 2001
 * Company:
 * @author Simon Greenwold
 * @version 1.0
 */
import java.awt.*;

public class TLineForcePoly extends TLine {

  public static final float ZERO_LENGTH = .3f;
  public static final int COMPRESSIVE = 1;
  public static final int TENSILE = -1;
  public static final int NONE = 0;

  public int mCharacter;

  public TPoint mMemberStart;
  public TPoint mMemberEnd;

  public static Color mColorCompressive;
  public static Color mColorTensile;
  public static Color mColorZero;

//  public G g;

  public TLineForcePoly() {//G aG) {
    mColorCompressive = G.mRed;
    mColorTensile = G.mBlue;
    mColorZero = G.mYellow;
    mSize = 3;
    //g = aG;
  }

  public void update() {
    super.update();
    if (length() <= ZERO_LENGTH) {
      mCharacter = NONE;
      mColor = mColorZero;
    }
    else if (Util.near(Util.direction(mMemberStart.x, mMemberStart.y, mMemberEnd.x, mMemberEnd.y),
        direction(), .02)) {
      mCharacter = TENSILE;
      mColor = mColorTensile;
    }
    else {
      mCharacter = COMPRESSIVE;
      mColor = mColorCompressive;
    }
  }
}